From d3e1c63b87842fec46f2ef77fdcd66983dc9f8ce Mon Sep 17 00:00:00 2001
From: Michael Banack <bob5972@banack.net>
Date: Mon, 14 Jun 2021 00:28:03 -0400
Subject: [PATCH 9/9] Fix errorTrap leak in free_win_data

The function free_win_data() may generate X11 errors as the client
window may already be gone.

To avoid those, the entire function code is supposed to be enclosed
within an error trap/push.

A typo in the code would however push twice instead of a balanced
push/pop which causes the list of event sequences in GDK to grow
indefinitely, leading to slowdowns as the list grows over time.

Fix the leak by balancing the error push with an error pop at the end
of the function.

xfce/xfwm4#351

(Source: https://gitlab.xfce.org/xfce/xfwm4/-/commit/e6a32cf7d789ca3058f50fd630de8e518d310cf8)
---
 src/compositor.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/compositor.c b/src/compositor.c
index 9177740..0050ab9 100644
--- a/src/compositor.c
+++ b/src/compositor.c
@@ -857,7 +857,7 @@ free_win_data (CWindow *cw, gboolean delete)
         cw->saved_picture = cw->picture;
         cw->picture = None;
     }
-    myDisplayErrorTrapPush (display_info);
+    myDisplayErrorTrapPopIgnore (display_info);
 }
 
 static Picture
-- 
2.35.3

